quiche/src/lib.rs

```diff
@@ -1974,17 +2020,33 @@ impl Default for QlogInfo {
 
 impl<F: BufFactory> Connection<F> {
     fn new(
-        scid: &ConnectionId, odcid: Option<&ConnectionId>, local: SocketAddr,
+        scid: &ConnectionId, dcid: Option<&ConnectionId>, local: SocketAddr,
         peer: SocketAddr, config: &mut Config, is_server: bool,
     ) -> Result<Connection<F>> {
         let tls = config.tls_ctx.new_handshake()?;
-        Connection::with_tls(scid, odcid, local, peer, config, tls, is_server)
+        Connection::with_tls(scid, dcid, local, peer, config, tls, is_server)
     }
 
     fn with_tls(
-        scid: &ConnectionId, odcid: Option<&ConnectionId>, local: SocketAddr,
+        scid: &ConnectionId, dcid: Option<&ConnectionId>, local: SocketAddr,
         peer: SocketAddr, config: &Config, tls: tls::Handshake, is_server: bool,
     ) -> Result<Connection<F>> {
+        if !is_server {
+            #[cfg(feature = "connect-with-dcid")]
+            if let Some(dcid) = dcid {
+                // The Minimum length is 8.
+                // See https://datatracker.ietf.org/doc/html/rfc9000#section-7.2-3
+                if dcid.to_vec().len() < 8 {
+                    return Err(Error::InvalidTransportParam);
+                }
+            }
+            #[cfg(not(feature = "connect-with-dcid"))]
+            // Just error out in case we did not compile with feature enabled to
+            // guard against miss-use in the future.
+            if dcid.is_some() {
+                return Err(Error::InvalidTransportParam);
```

Comment:
we could also panic in this case if this would be better ?

Reply:
This seems like it meant for extra precaution and doesnt seem necessary.

Reply:
sure that is fine with me as well... let me do this.